### DISCUSSION TITLE
Create new Matrix nested entry in a module

### DISCUSSION THREAD

**Discussion by shifuma** [craftcms/cms#15345] [Q&A] [answered]
I'm trying to programmatically create a nested entry inside a Matrix field:

```
$newEntry = new Entry();
$newEntry->typeId = 1;
$newEntry->title = 'My Title';
Craft::$app->getElements()->saveElement($newEntry);
```

But that gives me the error `Call to a member function getSiteSettings() on null`. I assume it's because I'm not passing a `sectionId`? I thought maybe setting the owner `$newEntry->ownerId = $owner->id;` might help, but the same issue.

**Comment by brandonkelly** [comment, marked as answer]
You need to _either_ set `sectionId` (for top-level entries) _or_ `fieldId` + `ownerId` (for nested entries).

**Reply by brandonkelly** [reply]
[Made that more clear](https://github.com/craftcms/cms/commit/89e8e937fcb8da8bb852152ad4b6b2463f9ba52e) for the next release. Going forward, instead of a PHP error, you’d get an exception thrown with the message:

```
Either `sectionId` or `fieldId` + `ownerId` must be set on the entry.
```
